### ISSUE TITLE
NumPy 1.20.0 incompatibilities

### ISSUE THREAD

**Issue by hvy** [optuna/optuna#2293]
## Motivation

NumPy released [v1.20.0](https://github.com/numpy/numpy/releases/tag/v1.20.0) in 21-01-31 with several updates including deprecations and type hint annotations. Currently, Optuna is not compatible with this version. Type hint errors are the known issues. The NumPy version in `setup.py` is therefore constrained to avoid this latest release with `<1.20.0` (https://github.com/optuna/optuna/pull/2292). These incompatibilities should be addressed and the version constraints should be removed. The drop of Python 3.6 should also be noted since Optuna supports it for the time being.

## Description

Fix incompatibilities with NumPy v.1.20.0.

## Alternatives (optional)

Type hint errors could be ignoring with `# type: ignore` to allow first relaxing the version constraint and introduce incremental fixes. 

### mypy errors/notes

- [ ] `optuna/importance/_fanova/_fanova.py`
- [ ] `optuna/importance/_fanova/_tree.py`
- [ ] `optuna/integration/botorch.py`
- [ ] `optuna/integration/cma.py`
- [ ] `optuna/integration/sklearn.py`
- [ ] `optuna/multi_objective/_hypervolume/wfg.py`
- [ ] `optuna/multi_objective/samplers/_motpe.py`
- [ ] `optuna/pruners/_percentile.py`
- [ ] `optuna/samplers/_tpe/multi_objective_sampler.py`
- [ ] `optuna/samplers/_tpe/multivariate_parzen_estimator.py`
- [ ] `optuna/samplers/_tpe/parzen_estimator.py`
- [ ] `optuna/samplers/_tpe/sampler.py`
- [ ] `tests/importance_tests/fanova_tests/test_tree.py`
- [ ] `tests/integration_tests/test_chainer.py`
- [ ] `tests/integration_tests/test_sklearn.py`
- [ ] `tests/multi_objective_tests/hypervolume_tests/test_wfg.py`
- [ ] `tests/multi_objective_tests/samplers_tests/test_motpe.py`
- [ ] `tests/samplers_tests/test_grid.py`
- [ ] `tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py`
- [ ] `tests/samplers_tests/tpe_tests/test_parzen_estimator.py`
- [ ] `tests/storages_tests/rdb_tests/test_with_server.py`

_Parsed from https://github.com/optuna/optuna/issues/2293#issuecomment-770624780_

**Comment by hvy** [comment]
mypy errors from CI (e.g. https://github.com/optuna/optuna/runs/1800022865?check_suite_focus=true)
```python
optuna/samplers/_tpe/parzen_estimator.py:83: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[int]")
optuna/samplers/_tpe/parzen_estimator.py:85: error: Incompatible types in assignment (expression has type "integer[Any]", variable has type "int")
optuna/samplers/_tpe/parzen_estimator.py:93: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[int]")
optuna/samplers/_tpe/parzen_estimator.py:128: error: Incompatible types in assignment (expression has type "Union[number[Any], ndarray]", variable has type "ndarray")
optuna/importance/_fanova/_tree.py:74: error: Incompatible types in assignment (expression has type "number[Any]", variable has type "float")
optuna/importance/_fanova/_tree.py:76: error: Item "ndarray" of "Union[List[float], ndarray]" has no attribute "append"
optuna/importance/_fanova/_tree.py:77: error: Item "ndarray" of "Union[List[float], ndarray]" has no attribute "append"
optuna/importance/_fanova/_tree.py:139: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[int]")
optuna/importance/_fanova/_tree.py:140: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[float]")
optuna/importance/_fanova/_tree.py:159: error: No overload variant of "__setitem__" of "list" matches argument types "int", "ndarray"
optuna/importance/_fanova/_tree.py:159: note: Possible overload variants:
optuna/importance/_fanova/_tree.py:159: note:     def __setitem__(self, int, None) -> None
optuna/importance/_fanova/_tree.py:159: note:     def __setitem__(self, slice, Iterable[None]) -> None
optuna/importance/_fanova/_tree.py:167: error: Argument 1 to "_get_cardinality" has incompatible type "None"; expected "ndarray"
optuna/importance/_fanova/_tree.py:172: error: Argument 2 to "_get_node_children_subspaces" of "_FanovaTree" has incompatible type "None"; expected "ndarray"
optuna/importance/_fanova/_tree.py:175: error: No overload variant of "__setitem__" of "list" matches argument types "int", "ndarray"
optuna/importance/_fanova/_tree.py:175: note: Possible overload variants:
optuna/importance/_fanova/_tree.py:175: note:     def __setitem__(self, int, None) -> None
optuna/importance/_fanova/_tree.py:175: note:     def __setitem__(self, slice, Iterable[None]) -> None
optuna/importance/_fanova/_tree.py:186: error: Incompatible types in assignment (expression has type "Union[number[Any], ndarray]", variable has type "float")
optuna/importance/_fanova/_tree.py:244: error: Incompatible return value type (got "ndarray", expected "List[Set[int]]")
tests/importance_tests/fanova_tests/test_tree.py:93: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[int]")
tests/importance_tests/fanova_tests/test_tree.py:94: error: Argument 1 to "get_marginal_variance" of "_FanovaTree" has incompatible type "List[int]"; expected "ndarray"
tests/importance_tests/fanova_tests/test_tree.py:210: error: Incompatible types in assignment (expression has type "bool", variable has type "ndarray")
optuna/multi_objective/_hypervolume/wfg.py:21: error: Incompatible types in assignment (expression has type "ndarray", variable has type "None")
optuna/multi_objective/_hypervolume/wfg.py:51: error: Argument 2 to "_compute_2points_volume" has incompatible type "None"; expected "ndarray"
optuna/multi_objective/_hypervolume/wfg.py:55: error: Argument 2 to "_compute_2points_volume" has incompatible type "None"; expected "ndarray"
optuna/importance/_fanova/_fanova.py:96: error: Item "ndarray" of "Union[List[float], ndarray]" has no attribute "append"
optuna/importance/_fanova/_fanova.py:100: error: Incompatible return value type (got "Tuple[number[Any], number[Any]]", expected "Tuple[float, float]")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:75: error: Incompatible types in assignment (expression has type "None", variable has type "ndarray")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:268: error: Incompatible types in assignment (expression has type "Union[number[Any], ndarray]", target has type "ndarray")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:271: error: Unsupported operand types for / ("ndarray" and "None")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:271: note: Right operand is of type "Optional[float]"
optuna/samplers/_tpe/multivariate_parzen_estimator.py:272: error: Incompatible types in assignment (expression has type "Union[number[Any], ndarray]", target has type "ndarray")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:277: error: Incompatible types in assignment (expression has type "Union[number[Any], ndarray]", target has type "ndarray")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:315: error: Unsupported operand types for / ("None" and "int")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:315: note: Left operand is of type "Optional[float]"
optuna/samplers/_tpe/multivariate_parzen_estimator.py:318: error: Unsupported operand types for / ("None" and "int")
optuna/samplers/_tpe/multivariate_parzen_estimator.py:318: note: Left operand is of type "Optional[float]"
optuna/samplers/_tpe/multivariate_parzen_estimator.py:363: error: Incompatible types in assignment (expression has type "Union[number[Any], ndarray]", variable has type "ndarray")
optuna/pruners/_percentile.py:18: error: Module has no attribute "float"; maybe "float_", "cfloat", or "float64"?
optuna/pruners/_percentile.py:48: error: Module has no attribute "float"; maybe "float_", "cfloat", or "float64"?
optuna/samplers/_tpe/sampler.py:327: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[Optional[float]]")
optuna/samplers/_tpe/sampler.py:328: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[Tuple[float, float]]")
optuna/samplers/_tpe/sampler.py:332: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna/samplers/_tpe/sampler.py:332: note: Possible overload variants:
optuna/samplers/_tpe/sampler.py:332: note:     def __getitem__(self, int) -> Optional[float]
optuna/samplers/_tpe/sampler.py:332: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna/samplers/_tpe/sampler.py:334: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna/samplers/_tpe/sampler.py:334: note: Possible overload variants:
optuna/samplers/_tpe/sampler.py:334: note:     def __getitem__(self, int) -> Optional[float]
optuna/samplers/_tpe/sampler.py:334: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna/samplers/_tpe/sampler.py:344: error: Value expression in dictionary comprehension has incompatible type "ndarray"; expected type "List[Optional[float]]"
optuna/samplers/_tpe/sampler.py:345: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[Tuple[float, float]]")
optuna/samplers/_tpe/sampler.py:355: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna/samplers/_tpe/sampler.py:355: note: Possible overload variants:
optuna/samplers/_tpe/sampler.py:355: note:     def __getitem__(self, int) -> Optional[float]
optuna/samplers/_tpe/sampler.py:355: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna/samplers/_tpe/sampler.py:356: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna/samplers/_tpe/sampler.py:356: note: Possible overload variants:
optuna/samplers/_tpe/sampler.py:356: note:     def __getitem__(self, int) -> Optional[float]
optuna/samplers/_tpe/sampler.py:356: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna/samplers/_tpe/sampler.py:485: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna/samplers/_tpe/sampler.py:486: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna/samplers/_tpe/multi_objective_sampler.py:467: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna/samplers/_tpe/multi_objective_sampler.py:468: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna/samplers/_tpe/multi_objective_sampler.py:520: error: No overload variant of "__setitem__" of "list" matches argument types "integer[Any]", "int"
optuna/samplers/_tpe/multi_objective_sampler.py:520: note: Possible overload variants:
optuna/samplers/_tpe/multi_objective_sampler.py:520: note:     def __setitem__(self, int, float) -> None
optuna/samplers/_tpe/multi_objective_sampler.py:520: note:     def __setitem__(self, slice, Iterable[float]) -> None
optuna/multi_objective/samplers/_motpe.py:474: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna/multi_objective/samplers/_motpe.py:475: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna/multi_objective/samplers/_motpe.py:527: error: No overload variant of "__setitem__" of "list" matches argument types "integer[Any]", "int"
optuna/multi_objective/samplers/_motpe.py:527: note: Possible overload variants:
optuna/multi_objective/samplers/_motpe.py:527: note:     def __setitem__(self, int, float) -> None
optuna/multi_objective/samplers/_motpe.py:527: note:     def __setitem__(self, slice, Iterable[float]) -> None
optuna/integration/sklearn.py:285: error: Argument 2 to "_store_scores" of "_Objective" has incompatible type "Dict[str, ndarray]"; expected "Dict[str, Union[List[float], ndarray, Any]]"
optuna/integration/sklearn.py:285: note: "Dict" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
optuna/integration/sklearn.py:285: note: Consider using "Mapping" instead, which is covariant in the value type
optuna/integration/sklearn.py:289: error: Incompatible return value type (got "Dict[str, ndarray]", expected "Dict[str, Union[List[float], ndarray, Any]]")
optuna/integration/sklearn.py:289: note: "Dict" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
optuna/integration/sklearn.py:289: note: Consider using "Mapping" instead, which is covariant in the value type
optuna/integration/sklearn.py:289: note: Perhaps you need a type annotation for "scores"? Suggestion: "Dict[str, Union[List[float], ndarray, Any]]"
optuna/integration/sklearn.py:669: error: Incompatible default for argument "error_score" (default has type "float", argument has type "Union[Number, str]")
optuna/integration/cma.py:242: error: Incompatible types in assignment (expression has type "int", target has type "number[Any]")
optuna/integration/cma.py:246: error: Incompatible types in assignment (expression has type "float", target has type "number[Any]")
optuna/integration/cma.py:249: error: Incompatible types in assignment (expression has type "Union[None, float, str]", target has type "number[Any]")
optuna/integration/botorch.py:494: error: "ndarray" has no attribute "dim"; maybe "ndim"?
optuna/integration/botorch.py:495: error: "ndarray" has no attribute "unsqueeze_"; maybe "squeeze"?
optuna/integration/botorch.py:496: error: "ndarray" has no attribute "transpose_"; maybe "transpose"?
optuna/integration/botorch.py:515: error: "int" not callable
optuna/integration/botorch.py:517: error: "int" not callable
optuna/integration/botorch.py:523: error: Incompatible types in assignment (expression has type "Dict[str, Any]", variable has type "ndarray")
optuna/integration/botorch.py:525: error: Incompatible return value type (got "ndarray", expected "Dict[str, Any]")
tests/storages_tests/rdb_tests/test_with_server.py:58: error: Argument 1 to "all" has incompatible type "Union[bool_, ndarray]"; expected "Iterable[object]"
tests/storages_tests/rdb_tests/test_with_server.py:59: error: Argument 1 to "isclose" has incompatible type "List[Optional[float]]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"
tests/storages_tests/rdb_tests/test_with_server.py:72: error: Argument 1 to "all" has incompatible type "Union[bool_, ndarray]"; expected "Iterable[object]"
tests/storages_tests/rdb_tests/test_with_server.py:79: error: Argument 1 to "all" has incompatible type "Union[bool_, ndarray]"; expected "Iterable[object]"
tests/samplers_tests/test_grid.py:164: error: Dict entry 0 has incompatible type "str": "ndarray"; expected "str": "Sequence[Union[str, float, int, bool, None]]"
tests/samplers_tests/tpe_tests/test_parzen_estimator.py:28: error: Argument 1 to "_calculate" of "_ParzenEstimator" has incompatible type "List[float]"; expected "ndarray"
tests/samplers_tests/tpe_tests/test_parzen_estimator.py:109: error: Argument 1 to "_calculate" of "_ParzenEstimator" has incompatible type "List[float]"; expected "ndarray"
tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py:621: error: Argument 2 to "_solve_hssp" of "MOTPESampler" has incompatible type "List[int]"; expected "ndarray"
tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py:635: error: Argument 2 to "_solve_hssp" of "MOTPESampler" has incompatible type "List[int]"; expected "ndarray"
tests/multi_objective_tests/samplers_tests/test_motpe.py:588: error: Argument 2 to "_solve_hssp" of "MOTPEMultiObjectiveSampler" has incompatible type "List[int]"; expected "ndarray"
tests/multi_objective_tests/samplers_tests/test_motpe.py:602: error: Argument 2 to "_solve_hssp" of "MOTPEMultiObjectiveSampler" has incompatible type "List[int]"; expected "ndarray"
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:24: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:26: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:36: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:38: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:48: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:49: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:52: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:52: error: List item 0 has incompatible type "List[ndarray]"; expected "Union[complex, generic, Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"
tests/multi_objective_tests/hypervolume_tests/test_wfg.py:55: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests/integration_tests/test_sklearn.py:292: error: Argument "error_score" to "OptunaSearchCV" has incompatible type "float"; expected "Union[Number, str]"
tests/integration_tests/test_chainer.py:29: error: Incompatible return value type (got "Tuple[ndarray, signedinteger[_32Bit]]", expected "Tuple[ndarray, int]")
Found 84 errors in 21 files (checked 240 source files)
```

**Comment by 0x41head** [comment]
Working on it .

**Comment by hvy** [comment]
That'd be great. Would you like to tackle all of them? We could split it up to allow incremental fixes. For instance, per directory/submodule and/or whether it's an error or a note (I was thinking of Tweeting this [here](https://twitter.com/OptunaAutoML/status/1351049129414324225) but though I'd check with you first).

**Comment by 0x41head** [comment]
Yes. That would be really helpful. I fixed some of it yesterday. So if anyone is solving it you can clone my master repo : 
https://github.com/0x41head/optuna/tree/master

This is the updated log :
`optuna\multi_objective\_hypervolume\wfg.py:21: error: Incompatible types in assignment (expression has type "ndarray", variable has type "None")
optuna\samplers\_tpe\multivariate_parzen_estimator.py:75: error: Incompatible types in assignment (expression has type "None", variable has type "ndarray")
optuna\samplers\_tpe\multivariate_parzen_estimator.py:271: error: Unsupported operand types for / ("ndarray" and "None")
optuna\samplers\_tpe\multivariate_parzen_estimator.py:271: note: Right operand is of type "Optional[float]"
optuna\samplers\_tpe\multivariate_parzen_estimator.py:315: error: Unsupported operand types for / ("None" and "int")
optuna\samplers\_tpe\multivariate_parzen_estimator.py:315: note: Left operand is of type "Optional[float]"
optuna\samplers\_tpe\multivariate_parzen_estimator.py:318: error: Unsupported operand types for / ("None" and "int")
optuna\samplers\_tpe\multivariate_parzen_estimator.py:318: note: Left operand is of type "Optional[float]"
optuna\samplers\_tpe\sampler.py:332: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna\samplers\_tpe\sampler.py:332: note: Possible overload variants:
optuna\samplers\_tpe\sampler.py:332: note:     def __getitem__(self, int) -> Optional[float]
optuna\samplers\_tpe\sampler.py:332: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna\samplers\_tpe\sampler.py:334: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna\samplers\_tpe\sampler.py:334: note: Possible overload variants:
optuna\samplers\_tpe\sampler.py:334: note:     def __getitem__(self, int) -> Optional[float]
optuna\samplers\_tpe\sampler.py:334: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna\samplers\_tpe\sampler.py:355: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna\samplers\_tpe\sampler.py:355: note: Possible overload variants:
optuna\samplers\_tpe\sampler.py:355: note:     def __getitem__(self, int) -> Optional[float]
optuna\samplers\_tpe\sampler.py:355: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna\samplers\_tpe\sampler.py:356: error: No overload variant of "__getitem__" of "list" matches argument type "ndarray"
optuna\samplers\_tpe\sampler.py:356: note: Possible overload variants:
optuna\samplers\_tpe\sampler.py:356: note:     def __getitem__(self, int) -> Optional[float]
optuna\samplers\_tpe\sampler.py:356: note:     def __getitem__(self, slice) -> List[Optional[float]]
optuna\samplers\_tpe\multi_objective_sampler.py:467: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna\samplers\_tpe\multi_objective_sampler.py:468: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna\samplers\_tpe\multi_objective_sampler.py:520: error: No overload variant of "__setitem__" of "list" matches argument types "integer[Any]", "int"
optuna\samplers\_tpe\multi_objective_sampler.py:520: note: Possible overload variants:
optuna\samplers\_tpe\multi_objective_sampler.py:520: note:     def __setitem__(self, int, float) -> None
optuna\samplers\_tpe\multi_objective_sampler.py:520: note:     def __setitem__(self, slice, Iterable[float]) -> None
optuna\multi_objective\samplers\_motpe.py:474: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna\multi_objective\samplers\_motpe.py:475: error: Incompatible types in assignment (expression has type "List[int]", variable has type "ndarray")
optuna\multi_objective\samplers\_motpe.py:527: error: No overload variant of "__setitem__" of "list" matches argument types "integer[Any]", "int"
optuna\multi_objective\samplers\_motpe.py:527: note: Possible overload variants:
optuna\multi_objective\samplers\_motpe.py:527: note:     def __setitem__(self, int, float) -> None
optuna\multi_objective\samplers\_motpe.py:527: note:     def __setitem__(self, slice, Iterable[float]) -> None
optuna\integration\sklearn.py:285: error: Argument 2 to "_store_scores" of "_Objective" has incompatible type "Dict[str, ndarray]"; expected "Dict[str, Union[List[float], ndarray, Any]]"
optuna\integration\sklearn.py:285: note: "Dict" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
optuna\integration\sklearn.py:285: note: Consider using "Mapping" instead, which is covariant in the value type
optuna\integration\sklearn.py:289: error: Incompatible return value type (got "Dict[str, ndarray]", expected "Dict[str, Union[List[float], ndarray, Any]]")
optuna\integration\sklearn.py:289: note: "Dict" is invariant -- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance
optuna\integration\sklearn.py:289: note: Consider using "Mapping" instead, which is covariant in the value type
optuna\integration\sklearn.py:289: note: Perhaps you need a type annotation for "scores"? Suggestion: "Dict[str, Union[List[float], ndarray, Any]]"
optuna\integration\sklearn.py:669: error: Incompatible default for argument "error_score" (default has type "float", argument has type "Union[Number, str]")
optuna\integration\cma.py:242: error: Incompatible types in assignment (expression has type "int", target has type "number[Any]")
optuna\integration\cma.py:246: error: Incompatible types in assignment (expression has type "float", target has type "number[Any]")
optuna\integration\cma.py:249: error: Incompatible types in assignment (expression has type "Union[None, float, str]", target has type "number[Any]")
optuna\integration\botorch.py:494: error: "ndarray" has no attribute "dim"; maybe "ndim"?
optuna\integration\botorch.py:495: error: "ndarray" has no attribute "unsqueeze_"; maybe "squeeze"?
optuna\integration\botorch.py:496: error: "ndarray" has no attribute "transpose_"; maybe "transpose"?
optuna\integration\botorch.py:515: error: "int" not callable
optuna\integration\botorch.py:517: error: "int" not callable
optuna\integration\botorch.py:523: error: Incompatible types in assignment (expression has type "Dict[str, Any]", variable has type "ndarray")
optuna\integration\botorch.py:525: error: Incompatible return value type (got "ndarray", expected "Dict[str, Any]")
tests\storages_tests\rdb_tests\test_with_server.py:58: error: Argument 1 to "all" has incompatible type "Union[bool_, ndarray]"; expected "Iterable[object]"
tests\storages_tests\rdb_tests\test_with_server.py:59: error: Argument 1 to "isclose" has incompatible type "List[Optional[float]]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"
tests\storages_tests\rdb_tests\test_with_server.py:72: error: Argument 1 to "all" has incompatible type "Union[bool_, ndarray]"; expected "Iterable[object]"
tests\storages_tests\rdb_tests\test_with_server.py:79: error: Argument 1 to "all" has incompatible type "Union[bool_, ndarray]"; expected "Iterable[object]"
tests\samplers_tests\test_grid.py:164: error: Dict entry 0 has incompatible type "str": "ndarray"; expected "str": "Sequence[Union[str, float, int, bool, None]]"
tests\samplers_tests\tpe_tests\test_parzen_estimator.py:28: error: Argument 1 to "_calculate" of "_ParzenEstimator" has incompatible type "List[float]"; expected "ndarray"
tests\samplers_tests\tpe_tests\test_parzen_estimator.py:109: error: Argument 1 to "_calculate" of "_ParzenEstimator" has incompatible type "List[float]"; expected "ndarray"
tests\samplers_tests\tpe_tests\test_multi_objective_sampler.py:621: error: Argument 2 to "_solve_hssp" of "MOTPESampler" has incompatible type "List[int]"; expected "ndarray"
tests\samplers_tests\tpe_tests\test_multi_objective_sampler.py:635: error: Argument 2 to "_solve_hssp" of "MOTPESampler" has incompatible type "List[int]"; expected "ndarray"
tests\multi_objective_tests\samplers_tests\test_motpe.py:588: error: Argument 2 to "_solve_hssp" of "MOTPEMultiObjectiveSampler" has incompatible type "List[int]"; expected "ndarray"
tests\multi_objective_tests\samplers_tests\test_motpe.py:602: error: Argument 2 to "_solve_hssp" of "MOTPEMultiObjectiveSampler" has incompatible type "List[int]"; expected "ndarray"
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:24: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:26: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:36: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:38: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:48: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:49: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:52: error: Incompatible types in assignment (expression has type "ndarray", variable has type "List[ndarray]")
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:52: error: List item 0 has incompatible type "List[ndarray]"; expected "Union[complex, generic, Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"
tests\multi_objective_tests\hypervolume_tests\test_wfg.py:55: error: Argument 1 to "compute" of "BaseHypervolume" has incompatible type "List[ndarray]"; expected "ndarray"
tests\integration_tests\test_sklearn.py:292: error: Argument "error_score" to "OptunaSearchCV" has incompatible type "float"; expected "Union[Number, str]"
tests\integration_tests\test_chainer.py:29: error: Incompatible return value type (got "Tuple[ndarray, signedinteger[_32Bit]]", expected "Tuple[ndarray, int]")
Found 50 errors in 16 files (checked 240 source files)`

**Comment by hvy** [comment]
Got it. Parsed the files and listed them up in the description. Maybe you could go ahead and create an initial PR and other contributors could push to your branch of create a new one based on it.

**Comment by Devanshu24** [comment]
I would be happy to help @0x41head !
Looking forward to your PR, so I can pitch in :)

**Comment by 0x41head** [comment]
@Devanshu24 @hvy done.

**Comment by 0x41head** [comment]
Issue resolved with PR #2300.Can be closed.

**Comment by hvy** [comment]
Thanks for the heads up. Let me close the issue.
